documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,303 @@
+We will be building a REST application that is using DataStax Object Mapper. It will simplify the Data Access Layer code.
+
+The most important part of the `pom.xml` is adding the `cassandra-quarkus` extension:
+[source,xml]
+----
+<dependency>
+  <groupId>com.datastax.oss.quarkus</groupId>
+  <artifactId>cassandra-quarkus-client</artifactId>
+  <version>${quarkus.version}</version>
+</dependency>
+----
+
+== Creating JSON REST service
+In this example, we will create an application to manage a list of fruits.
+
+First, let's create the `Fruit` bean as follows:
+[source,java]
+----
+@Entity
+public class Fruit {
+
+  @ClusteringColumn private String name;
+  private String description;
+  @PartitionKey private String id;
+
+  public Fruit() {}
+
+  public Fruit(String name, String description, String id) {
+    this.name = name;
+    this.description = description;
+    this.id = id;
+  }
+  // getters, setters, hashCode and equals omitted for brevity
+}
+----
+We are using ObjectMapper, due to that fact the class is annotated with an `@Entity`.
+Also, the Fruit `id` is a `@PartitionKey` and `name` is a `@ClusteringColumn` so we are using the corresponding annotations
+from the ObjectMapper. It will allow the Mapper to generate proper CQL queries underneath.
+
+One important thing to note is that having a default constructor is required by the Object Mapper.
+
+To leverage the Mapper logic in this app we need to create a DAO:
+[source,java]
+----
+@Dao
+public interface FruitDao {
+  @Update
+  void update(Fruit fruit);
+
+  @Select
+  PagingIterable<Fruit> findById(String id);
+}
+----
+
+This class exposes operations that will be used in the REST service.
+
+Finally, the Mapper itself:
+[source,java]
+----
+@Mapper
+public interface FruitMapper {
+  @DaoFactory
+  FruitDao fruitDao(@DaoKeyspace CqlIdentifier keyspace);
+}
+----
+It is constructing the `FruitDao`.
+
+Now create a `com.datastax.oss.quarkus.FruitService` that will be the business layer of our application and store/load the fruits from the Cassandra database.
+[source,java]
+----
+@ApplicationScoped
+public class FruitService {
+  private final FruitDao dao;
+
+  @Inject
+  public FruitService(CqlSession session) {
+    dao = new FruitMapperBuilder(session).build().fruitDao(CqlIdentifier.fromCql("k1"));
+  }
+
+  public void save(Fruit fruit) {
+    dao.update(fruit);
+  }
+
+  public List<Fruit> get(String id) {
+    return dao.findById(id).all();
+  }
+}
+----
+Please note the construction of dao. It is using `CqlSession` injected automatically by the cassandra-quarkus extension.
+
+The last missing piece is the REST API that will expose GET and POST methods:
+[source,java]
+----
+@Path("/fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FruitResource {
+
+  private static final String STORE_NAME = "acme";
+  @Inject FruitService fruitService;
+
+  @GET
+  public List<Fruit> list() {
+    return fruitService.get(STORE_NAME);
```

Comment:
I definitely don't understand this data model. Isn't `id` supposed to be the fruit id?

Reply:
It is listing all fruits withing STORE_ID partition key.